### PR TITLE
[OrderStates] skip portfolio available amount up on exchange orders init

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
fixed negative portfolio error on initial order fetching

### CODE REVIEW

File: octobot_trading/personal_data/orders/channel/orders.py [hunk 1]
```diff
@@ -23,10 +23,10 @@
 
 
 class OrdersProducer(exchanges_channel.ExchangeChannelProducer):
-    async def push(self, orders, is_from_bot=False, are_closed=False):
-        await self.perform(orders, is_from_bot=is_from_bot, are_closed=are_closed)
+    async def push(self, orders, is_from_bot=False, are_closed=False, is_from_init=False):
+        await self.perform(orders, is_from_bot=is_from_bot, are_closed=are_closed, is_from_init=is_from_init)
 
-    async def perform(self, orders, is_from_bot=False, are_closed=False):
+    async def perform(self, orders, is_from_bot=False, are_closed=False, is_from_init=False):
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/channel/orders.py:29]: Why not using `is_from_bot` ?

Comment (GuillaumeDSM) [reply octobot_trading/personal_data/orders/channel/orders.py:29]: There might be order that are not form the bot but that are created after the bot started, in this case we need to update the portfolio available amounts

File: octobot_trading/personal_data/orders/states/open_order_state.py [hunk 2]
```diff
@@ -19,8 +19,10 @@
 
 
 class OpenOrderState(order_state.OrderState):
-    def __init__(self, order, is_from_exchange_data):
+    def __init__(self, order, is_from_exchange_data, is_from_exchange_init=False):
         super().__init__(order, is_from_exchange_data)
+        self.is_from_exchange_init = is_from_exchange_init
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/states/open_order_state.py:24]: Why not adding this attribute to `Order` class? It should be easier I think.

Comment (Herklos) [reply octobot_trading/personal_data/orders/states/open_order_state.py:24]: We wouldn't need to modify all order the order channel.

File: octobot_trading/personal_data/orders/channel/orders_updater.py [hunk 3]
```diff
@@ -98,11 +98,12 @@ async def _open_orders_fetch_and_push(self, is_from_bot=True, limit=ORDERS_UPDAT
...
             if open_orders:
                 await self.push(orders=list(map(self.channel.exchange_manager.exchange.clean_order, open_orders)),
-                                is_from_bot=is_from_bot)
+                                is_from_bot=is_from_bot,
+                                is_from_init=is_from_init)
             else:
                 await self.handle_post_open_order_update(symbol, open_orders, False)
 
-    async def _closed_orders_fetch_and_push(self, limit=ORDERS_UPDATE_LIMIT) -> None:
+    async def _closed_orders_fetch_and_push(self, limit=ORDERS_UPDATE_LIMIT, is_from_init=False) -> None:
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/channel/orders_updater.py:106]: Why ?

Comment (GuillaumeDSM) [reply octobot_trading/personal_data/orders/channel/orders_updater.py:106]: to be able to forward the info in orders channel

File: octobot_trading/personal_data/orders/channel/orders_updater.py [hunk 4]
```diff
@@ -63,6 +63,8 @@ async def initialize(self) -> None:
             await self.pause()
         except Exception as e:
             self.logger.error(f"Fail to initialize orders : {e}")
+        finally:
+            self.channel.exchange_manager.exchange_personal_data.orders_manager.are_exchange_orders_initialized = True
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/channel/orders_updater.py:67]: :+1:

File: octobot_trading/personal_data/orders/channel/orders_updater_simulator.py [hunk 5]
```diff
@@ -22,6 +22,7 @@
 
 class OrdersUpdaterSimulator(orders_updater.OrdersUpdater):
     async def start(self):
+        self.channel.exchange_manager.exchange_personal_data.orders_manager.are_exchange_orders_initialized = True
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/channel/orders_updater_simulator.py:25]: :+1:

**Review by Herklos — APPROVED** [review]
:+1:

**Pull request merged**